main/auth/twitch.py

```diff
@@ -0,0 +1,70 @@
+# coding: utf-8
+
+from __future__ import absolute_import
+
+import flask
+
+import auth
+import config
+import model
+import util
+
+from main import app
+
+twitch_config = dict(
+  access_token_method='POST',
+  access_token_url='https://id.twitch.tv/oauth2/token',
+  api_base_url='https://api.twitch.tv/helix/',
+  authorize_url='https://id.twitch.tv/oauth2/authorize',
+  client_id=config.CONFIG_DB.twitch_client_id,
+  client_secret=config.CONFIG_DB.twitch_client_secret,
+  client_kwargs={
+    'scope': 'user:read:email',
+    'token_endpoint_auth_method': 'client_secret_post',
+		'Client-ID': config.CONFIG_DB.twitch_client_id,
+  },
+)
+
+twitch = auth.create_oauth_app(twitch_config, 'twitch')
+
+def twitch_compliance_fix(url, headers, data):
+  headers["Client-ID"] = config.CONFIG_DB.twitch_client_id
+  return url, headers, data
+
+twitch.pre_request = twitch_compliance_fix
+
+@app.route('/api/auth/callback/twitch/')
+def twitch_authorized():
+  id_token = twitch.authorize_access_token()
+  if id_token is None:
+    flask.flash('You denied the request to sign in.')
+    return flask.redirect(util.get_next_url())
+
+  me = twitch.get('user')
```

Comment:
Actually the problem is here.. there is no `user` in `twitch`. How can we list what's there?

Reply:
I think the endpoint is “users”. You can learn it from loginpass